src/main/java/com/cleanroommc/groovyscript/compat/vanilla/Commands.java

```diff
@@ -0,0 +1,99 @@
+package com.cleanroommc.groovyscript.compat.vanilla;
+
+import com.cleanroommc.groovyscript.GroovyScript;
+import com.cleanroommc.groovyscript.api.GroovyBlacklist;
+import com.cleanroommc.groovyscript.api.IScriptReloadable;
+import com.cleanroommc.groovyscript.command.SimpleCommand;
+import com.cleanroommc.groovyscript.core.mixin.CommandHandlerAccessor;
+import com.cleanroommc.groovyscript.registry.AbstractReloadableStorage;
+import com.cleanroommc.groovyscript.registry.NamedRegistry;
+import net.minecraft.command.CommandHandler;
+import net.minecraft.command.ICommand;
+import net.minecraft.server.MinecraftServer;
+import net.minecraftforge.client.ClientCommandHandler;
+import net.minecraftforge.fml.common.FMLCommonHandler;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.Consumer;
+
+public class Commands extends NamedRegistry implements IScriptReloadable {
+
+    private final List<ICommand> serverCommands = new ArrayList<>();
+    private final AbstractReloadableStorage<ICommand> serverReloadableCommands = new AbstractReloadableStorage<>();
+    private final AbstractReloadableStorage<ICommand> clientReloadableCommands = new AbstractReloadableStorage<>();
+    private boolean serverStarted = false;
+
+    public void registerCommand(ICommand command) {
+        if (GroovyScript.getSandbox().isRunning() && GroovyScript.getSandbox().getCurrentLoader().isReloadable()) {
+            this.serverReloadableCommands.addScripted(command);
+        } else {
+            this.serverCommands.add(command);
+        }
+        if (this.serverStarted) {
+            forServer(commandHandler -> commandHandler.registerCommand(command));
+        }
+    }
+
+    public void registerClientCommand(ICommand command) {
+        if (FMLCommonHandler.instance().getSide().isServer()) return;
```

Comment:
should this print a warning?

Reply:
No. This way you can just register a command  without having to make sure it doent run on server